lib/adskHydraSceneBrowser/test/adskHydraSceneBrowserTestFixture.cpp

```diff
@@ -90,6 +91,60 @@ void AdskHydraSceneBrowserTestFixture::SetReferenceSceneIndex(
     sceneIndex = referenceSceneIndex;
 }
 
+void AdskHydraSceneBrowserTestFixture::VerifySceneCorrectness()
+{
+    // Traverse prim hierarchy
+    QTreeWidgetItemIterator itPrimsTreeWidget = GetIteratorForTree(_primHierarchyWidget);
```

Comment:
Thanks for the test, looks good, but it feels out of place to tie it to the scene browser.  I asked @debloip-adsk and he agreed with me, saying that 
> with APIs like GetChildPrimPaths, GetPrim, Get methods for container data sources (or GetElement for vector data sources) + casts, and GetValue/GetTypeValue methods on data sources, the whole scene index scene hierarchy can be inspected.

Here's an example you might like:
https://github.com/Autodesk/maya-hydra/blob/dev/test/lib/mayaUsd/render/mayaToHydra/cpp/testPrimInstancing.cpp
It shouldn't change the logic of the test.